lib/pdf_fill/forms/va21p527ez.rb

```diff
@@ -0,0 +1,695 @@
+        end
+
+        @form_data
+      end
+
+      def cont_merge_fields
+        @form_data['vaHospitalTreatments'].tap do |va_hospital_treatments|
+          @form_data['vaHospitalTreatmentNames'] = combine_va_hospital_names(va_hospital_treatments)
+          @form_data['vaHospitalTreatmentDates'] = rearrange_hospital_dates(
+            combine_va_hospital_dates(va_hospital_treatments)
+          )
+        end
+        @form_data.delete('vaHospitalTreatments')
+
+        @form_data['disabilityNames'] = get_disability_names(@form_data['disabilities'])
+
+        @form_data['cityState'] = combine_city_state(@form_data['veteranAddress'])
+        @form_data['veteranAddressLine1'] = combine_address(@form_data['veteranAddress'])
+        @form_data.delete('veteranAddress')
+
+        @form_data['previousNames'] = combine_previous_names(@form_data['previousNames'])
+
+        combine_name_addr(@form_data['nationalGuard'])
+
+        @form_data['jobs'] = rearrange_jobs(@form_data['jobs'])
+
+        %w(activeServiceDateRange powDateRange).each do |attr|
+          expand_date_range(@form_data, attr)
+        end
+
+        cont_merge_fields2
+      end
+
+      def cont_merge_fields2
+        @form_data['jobs'].tap do |jobs|
+          next if jobs.blank?
+
+          jobs.each do |job|
+            expand_date_range(job, 'dateRange')
+          end
+        end
+
+        expand_children(@form_data, 'children')
+
+        %w(marriages spouseMarriages).each do |marriage_type|
+          expand_marriages(@form_data, marriage_type)
+        end
+
+        @form_data['spouseMarriageCount'] = @form_data['spouseMarriages']&.length
+        @form_data['marriageCount'] = @form_data['marriages']&.length
+
+        @form_data['spouseAddress'] = combine_full_address(@form_data['spouseAddress'])
+
+        expand_marital_status(@form_data, 'maritalStatus')
+
+        expand_expected_incomes
+        expand_net_worths
+        expand_monthly_incomes
+
+        expand_bank_acct(@form_data['bankAccount'])
+
+        @form_data
+      end
+
+      def replace_phone_fields
+        %w(nightPhone dayPhone mobilePhone).each do |attr|
+          replace_phone(@form_data, attr)
+        end
+        replace_phone(@form_data['nationalGuard'], 'phone')
+      end
+
+      def merge_fields
+        @form_data['veteranFullName'] = combine_full_name(@form_data['veteranFullName'])
+
+        %w(
+          gender
+          vaFileNumber
+          previousNames
+          severancePay
+          powDateRange
+        ).each do |attr|
+          @form_data.merge!(public_send("expand_#{attr.underscore}", @form_data[attr]))
+        end
+
+        %w(
+          nationalGuardActivation
+          combatSince911
+          spouseIsVeteran
+          liveWithSpouse
+        ).each do |attr|
+          @form_data.merge!(public_send('expand_chk_and_del_key', @form_data, attr))
+        end
+
+        replace_phone_fields
+
+        cont_merge_fields
```

Comment:
Because these `cont_merge` methods are called serially, I'd just bite the bullet, set a rubocop exception, and put them all in this `merge fields` method. With any luck as we add more PDFs the logic can be pulled out into different modules and methods.